src/compiler/compiler.rs

```diff
@@ -1107,18 +1105,49 @@ __VERSION__
         }
     });
     if let Some(kind) = lines.next() {
-        let version = lines
-            .next()
-            // In case the compiler didn't expand the macro.
-            .filter(|&line| line != "__VERSION__")
-            .map(str::to_owned);
+        // Unescape the version string. We only remove surrounding quotes assuming that the version string does not
+        // contain characters that needed escaping. If the compiler did not know the macro (it is a gcc extension and
+        // not supported by eg. msvc), the line will still contain __VERSION__ and we have no version information.
+        // In case the macro was expanded to something other than a quoted string, still assume that it represents
+        // the compiler version.
+        let version = match lines.next() {
+            Some("__VERSION__") => None,
+            Some(s) if s.len() >= 2 && s.starts_with('\"') && s.ends_with('\"') => {
+                Some(s[1..s.len() - 1].to_owned())
+            }
+            Some(s) if s.trim().is_empty() => None,
+            Some(s) => Some(s.to_owned()),
+            None => None,
+        };
         match kind {
             "clang" | "clang++" | "apple-clang" | "apple-clang++" => {
                 debug!("Found {}", kind);
+
+                // Check whether the compiler also supports the -E -fminimize-whitespace flag
```

Comment:
Please move this into a separate function and add a test to verify it won't regress